VirtoCommerce.MarketingModule.Web/Scripts/promotion/blades/promotion-expressions.js

```diff
@@ -22,52 +22,165 @@ angular.module('virtoCommerce.marketingModule')
+                        parentBlade.error = 'Must select Product';
+                        listItem.selected = undefined;
+                    } else {
+                        if (isSelected) {
+                            if (_.all(parentBlade.selectedListEntries, function (x) { return x.id != listItem.id; })) {
+                                parentBlade.selectedListEntries.push(listItem);
+                            }
+                        }
+                        else {
+                            parentBlade.selectedListEntries = _.reject(parentBlade.selectedListEntries, function (x) { return x.id == listItem.id; });
+                        }
+                        parentBlade.error = undefined;
+                    }
+                }
+            };
+
+            bladeNavigationService.showBlade(catalogBlade, parentBlade);
+        };
 
         var selectedListEntries = [];
         var newBlade = {
-            id: "CatalogEntrySelect",
-            title: "marketing.blades.catalog-items-select.title-product",
-            controller: 'virtoCommerce.catalogModule.catalogItemSelectController',
-            template: 'Modules/$(VirtoCommerce.Catalog)/Scripts/blades/common/catalog-items-select.tpl.html',
+            id: "CatalogConditions",
+            title: "marketing.blades.catalog-conditions.title-product",
+            controller: 'virtoCommerce.marketingModule.catalogConditionsController',
+            template: 'Modules/$(VirtoCommerce.Marketing)/Scripts/promotion/blades/catalog-conditions.tpl.html',
             breadcrumbs: [],
             toolbarCommands: [
             {
-                name: "platform.commands.pick-selected", icon: 'fa fa-plus',
+                name: "platform.commands.refresh", icon: 'fa fa-refresh',
                 executeMethod: function (blade) {
-                    //parentElement.selectedListEntry = selectedListEntries[0];
-                    parentElement.productId = selectedListEntries[0].id;
-                    parentElement.productName = selectedListEntries[0].name;
-                    parentElement.productCode = selectedListEntries[0].code;
-                    bladeNavigationService.closeBlade(blade);
+                    blade.refresh();
+                },
+                canExecuteMethod: function () { return true; }
+            },
+            {
+                name: "platform.commands.add", icon: 'fa fa-plus',
+                executeMethod: function (blade) {
+                    openCatalogBlade(blade, newBlade);
                 },
                 canExecuteMethod: function () {
-                    return selectedListEntries.length == 1;
+                    return true;
                 }
-            }]
-        };
+            },
+            {
+                name: "platform.commands.delete", icon: 'fa fa-trash-o',
+                executeMethod: function (blade) {
+                    var grid = blade.$scope.gridApi.grid;
```

Comment:
Probably better to make all work with grid in blade were grid is located and created.